homeassistant/components/automation/device.py

```diff
@@ -0,0 +1,30 @@
+"""Offer device oriented automation."""
+import importlib
+import voluptuous as vol
+
+from homeassistant.const import CONF_DOMAIN, CONF_PLATFORM
+from homeassistant.loader import async_get_integration
+
+
+def _domain_validator(config):
+    """Validate it is a valid  domain or platform."""
+    try:
+        platform = importlib.import_module(
```

Comment:
This is really ugly, and it won't work with custom integrations either.  We should use the `async_get_integration` method instead. However, that's async. Maybe just drop this validator ?

Reply:
Yeah, the problem was that `async_get_integration` is async. I'll remove the validator for now.